### PR TITLE
Feature/migrate modeling from filesystem to database

### PR DESCRIPTION
**Pull request by SiggZ**
### Checklist
- [x] I run `yarn run webpack:build:main`: the project builds without errors.
- [x] I run `yarn lint`: there project builds without code style warnings.
- [x] ~I removed unnecessary whitespace changes in all related files.~
- [x] I updated the documentation and models.
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] I added (end-to-end) test cases for the new functionality.

### Motivation and Context
The UML models and modeling assessments were stored in the file system instead of the database. Therefore, the data model described in https://github.com/ls1intum/ArTEMiS#data-model did not apply to modeling exercises.

### Description
This PR migrates UML models and modeling assessments from the file system to the database. It also changes the data model of modeling assessment to be compliant to the data model described in https://github.com/ls1intum/ArTEMiS#data-model, i.e. modeling assessments are represented by a Feedback List contained in a Result for the corresponding modeling submission.

### Steps for Testing
1. Log in to ArTEMiS as a student
2. Navigate to the course overview
3. Save/Submit a submission for a modeling exercise
4. Check database for stored model of the submitted modeling submission
5. Log in to ArTEMiS as a tutor
6. Navigate to assessment dashboard
7. Submit an assessment for the previously submitted submission
8. Check database for stored Result and Feedback entries
9. In general, play around with modeling submissions and assessments (also automatic assessments) and check, if everything works as expected.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
+#Sun Feb 24 14:04:22 CET 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-all.zip
```
Comment (krusche) [inline gradle/wrapper/gradle-wrapper.properties]: Please revert the changes in this file, they are not necessary and unrelated to the changes. It’s the second time that we have them in a pull request

File: src/main/java/de/tum/in/www1/artemis/domain/Submission.java [hunk 2]
```diff
@@ -69,6 +69,13 @@
     @Column(name = "submission_date")
     private ZonedDateTime submissionDate;
 
+    public Submission() {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/Submission.java]: Please remove these constructors. We don’t need them and it’s unclear if they influence some behavior in hibernate or Jackson

File: src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java [hunk 3]
```diff
@@ -15,4 +15,6 @@
 @Repository
 public interface FeedbackRepository extends JpaRepository<Feedback, Long> {
     List<Feedback> findByResult(Result result);
+
+    void deleteByResult(Result result);
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java]: I think we don’t need this anymore, right? Then please delete it

File: src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java [hunk 4]
```diff
@@ -94,77 +102,63 @@ public ModelingSubmission save(ModelingSubmission modelingSubmission, ModelingEx
         return modelingSubmission;
     }
 
+
+    public void setResult (ModelingSubmission submission){
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java]: Please add javadoc

File: src/main/java/de/tum/in/www1/artemis/service/ResultService.java [hunk 5]
```diff
@@ -55,6 +55,13 @@ public Result findOneWithSubmission(long id) {
             .orElseThrow(() -> new EntityNotFoundException("Result with id: \"" + id + "\" does not exist"));
     }
 
+    public void setAssessor(Result result){
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/ResultService.java]: Javadoc is missing

File: src/main/java/de/tum/in/www1/artemis/service/ResultService.java [hunk 6]
```diff
@@ -55,6 +55,13 @@ public Result findOneWithSubmission(long id) {
...
+        User currentUser = userService.getUserWithGroupsAndAuthorities();
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/ResultService.java]: I think you can get the user easier, e.g. with the principal. This might include a database call

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (jmaximilian2) [reply gradle/wrapper/gradle-wrapper.properties]: changes have been reverted

File: src/main/java/de/tum/in/www1/artemis/domain/Submission.java (hunk 2 above)
Comment (jmaximilian2) [reply src/main/java/de/tum/in/www1/artemis/domain/Submission.java]: have been removed

File: src/main/java/de/tum/in/www1/artemis/service/ResultService.java (hunk 6 above)
Comment (jmaximilian2) [reply src/main/java/de/tum/in/www1/artemis/service/ResultService.java]: I did take a look into this and the Principle i get injected in the Controller does not include our User database object. An alternative using the Principle would avoid specifically looking up the current logged in user in the SpringSecurityContext but still require a database query to find that User if im not mistaken.

File: src/main/java/de/tum/in/www1/artemis/service/ResultService.java (hunk 5 above)
Comment (jmaximilian2) [reply src/main/java/de/tum/in/www1/artemis/service/ResultService.java]: javadoc got added

File: src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java (hunk 4 above)
Comment (jmaximilian2) [reply src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java]: javadoc got added

File: src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java (hunk 3 above)
Comment (SiggZ) [reply src/main/java/de/tum/in/www1/artemis/repository/FeedbackRepository.java]: done

File: src/main/webapp/app/entities/modeling-assessment/modeling-assessment.service.ts [hunk 7]
```diff
@@ -71,30 +76,21 @@ export class ModelingAssessmentService {
...
-    getNamesForAssessments(assessments: ModelingAssessment[], model: State): Map<string, Map<string, string>> {
+    getNamesForAssessments(result: Result, model: State): Map<string, Map<string, string>> {
         const assessmentsNames = new Map<string, Map<string, string>>();
-        for (const assessment of assessments) {
-            if (assessment.type === ModelElementType.CLASS) {
-                const classElement = model.entities.byId[assessment.id];
+        for (const feedback of result.feedbacks) {
+            const referencedModelType = feedback.referenceType;
+            const referencedModelId = feedback.referenceId;
+            if (feedback.reference === ModelElementType.CLASS) {
```
Comment (SiggZ) [inline src/main/webapp/app/entities/modeling-assessment/modeling-assessment.service.ts]: I think the referencedModelType should be checked here instead of the full reference: `if (referencedModelType === ModelElementType.CLASS) {` 
Or am I missing something?

Comment (krusche) [reply src/main/webapp/app/entities/modeling-assessment/modeling-assessment.service.ts]: yes, please fix it directly

Comment (SiggZ) [reply src/main/webapp/app/entities/modeling-assessment/modeling-assessment.service.ts]: done
